core/src/main/java/io/questdb/griffin/engine/join/LtJoinNoKeyFastRecordCursorFactory.java

```diff
@@ -0,0 +1,362 @@
+        private final OuterJoinRecord record;
+        private final int slaveTimestampIndex;
+        private boolean isMasterHasNextPending;
+        // stands for forward/backward scan through slave's time frames
+        private boolean isSlaveForwardScan;
+        private boolean isSlaveOpenPending;
+        private RecordCursor masterCursor;
+        private boolean masterHasNext;
+        private Record masterRecord;
+        private TimeFrameRecordCursor slaveCursor;
+        private int slaveFrameIndex = -1;
+        private long slaveFrameRow = Long.MIN_VALUE;
+        private Record slaveRecA;
+        private Record slaveRecB;
+        private long slaveTimestamp = Long.MIN_VALUE;
+
+        public AsOfJoinFastRecordCursor(
+                int columnSplit,
+                Record nullRecord,
+                int masterTimestampIndex,
+                int slaveTimestampIndex,
+                int lookahead
+        ) {
+            this.columnSplit = columnSplit;
+            this.record = new OuterJoinRecord(columnSplit, nullRecord);
+            this.masterTimestampIndex = masterTimestampIndex;
+            this.slaveTimestampIndex = slaveTimestampIndex;
+            this.lookahead = lookahead;
+        }
+
+        @Override
+        public void calculateSize(SqlExecutionCircuitBreaker circuitBreaker, Counter counter) {
+            masterCursor.calculateSize(circuitBreaker, counter);
+        }
+
+        @Override
+        public void close() {
+            masterCursor = Misc.free(masterCursor);
+            slaveCursor = Misc.free(slaveCursor);
+        }
+
+        @Override
+        public Record getRecord() {
+            return record;
+        }
+
+        @Override
+        public SymbolTable getSymbolTable(int columnIndex) {
+            if (columnIndex < columnSplit) {
+                return masterCursor.getSymbolTable(columnIndex);
+            }
+            return slaveCursor.getSymbolTable(columnIndex - columnSplit);
+        }
+
+        @Override
+        public boolean hasNext() {
+            if (isMasterHasNextPending) {
+                masterHasNext = masterCursor.hasNext();
+                isMasterHasNextPending = false;
+            }
+            if (masterHasNext) {
+                final long masterTimestamp = masterRecord.getTimestamp(masterTimestampIndex);
+                if (masterTimestamp <= slaveTimestamp) {
+                    isMasterHasNextPending = true;
+                    return true;
+                }
+                nextSlave(masterTimestamp);
+                isMasterHasNextPending = true;
+                return true;
+            }
+            return false;
+        }
+
+        @Override
+        public SymbolTable newSymbolTable(int columnIndex) {
+            if (columnIndex < columnSplit) {
+                return masterCursor.newSymbolTable(columnIndex);
+            }
+            return slaveCursor.newSymbolTable(columnIndex - columnSplit);
```

Comment:
nit: not covered by tests

Reply:
Added a test to cover this branch in 4c3b75023bb6ce4df54a4e1ba6c2a8488da0755a